### PR TITLE
Closes #5149 - remove wpr options from not public post types

### LINKED ISSUE #5149: Remove WP Rocket's options from not public post types
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version
- Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
When the post type is not `public` we should not display:
1. `Clear this cache` button on the view listing all CPT entries
2. `Clear This Post` button from top menu while editing the post
3. `Purge this URL` button from the top menu while viewing the post when logged-in as admin

Per:
https://github.com/wp-media/wp-rocket/issues/5010#issuecomment-1157295620

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by jeawhanlee**
## Description

This PR removes WPR options on non public post types.

Fixes #5149 

## Type of change
- [x] Enhancement (non-breaking change which improves an existing functionality)

# Checklist:

Please delete the options that are not relevant.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

File: inc/admin/admin.php [hunk 1]
```diff
@@ -67,7 +67,16 @@ function rocket_post_row_actions( $actions, $post ) {
...
 
-	if ( apply_filters( 'rocket_skip_post_row_actions', false, $post ) ) {
+	$cpts = get_post_types(
+		[
+			'public' => true,
+		],
+		'objects'
+	);
+
+	$cpts = apply_filters( 'rocket_skip_post_row_actions', $cpts );
```
Comment (remyperona) [inline inc/admin/admin.php:89]: Add docblock for the filter

File: inc/admin/ui/meta-boxes.php [hunk 2]
```diff
@@ -10,8 +10,20 @@
...
-		printf( '<div id="purge-action"><a class="button-secondary" href="%s">%s</a></div>', esc_url( $url ), esc_html__( 'Clear cache', 'rocket' ) );
+
+		$cpts = get_post_types(
+			[
+				'public' => true,
+			],
+			'objects'
+		);
+
+		$cpts = apply_filters( 'rocket_submitbox_options_post_types', $cpts );
```
Comment (remyperona) [inline inc/admin/ui/meta-boxes.php:33]: Add docblock here too

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Comment by Mai-Saad** [comment]
@jeawhanlee Can see a Clear cache option for the draft post list, at the admin bar of the edit draft post, and, if the classic editor is used, can see a clear cache option inside the page edit view. Can you please check?





@piotrbak  shouldn't we remove the option from the trash post list?

**Review by mostafa-hisham — APPROVED** [review]

**Comment by Mai-Saad** [comment]
@jeawhanlee Thanks for the update, I can see most of the cases working now. only the ADD new page now while using classic editor contains a clear cache option (page/post still not saved as draft nor published yet), can we handle this case?
Screenshot from 2022-09-07 18-38-01

Note: there is a warning on the branch, still under investigation to get consistent reproducible steps (started discussion [here](https://wp-media.slack.com/archives/GUT7FLHF1/p1662572541730159) )

**Comment by Mai-Saad** [comment]
Working fine to clear the cache. Remain only, removing clear used CSS of this URL for the non-public post in preview or view page + checking the same case for Elementor template view @jeawhanlee can you please check
Note: update for test cases is needed
[testrail-report-410.pdf](https://github.com/wp-media/wp-rocket/files/9535050/testrail-report-410.pdf)

**Comment by Mai-Saad** [comment]
@jeawhanlee can we handle Clear used of this URL not showing for private pages on the same PR?

**Comment by Mai-Saad** [comment]
What's remaining to be handled:
scenario1:
1- create any post
2- from quick edit, check private
3- from the posts list click view for this post
4- hover on the admin bar

Scenario 2:
1- Create an Elementor template (published one)
2- click preview from the template list
3- hover the admin bar

**Review by Mai-Saad — APPROVED** [review]
Working fine now for Elementor/ WC smart coupon templates, private, trashed/deleted, and draft posts

**Pull request merged**